include/internal/libfreenect2/async_packet_processor.h

```diff
@@ -119,6 +135,7 @@ class AsyncPacketProcessor : public PacketProcessor<PacketT>
       {
         // invoke process impl
         processor_->process(current_packet_);
+        releaseBuffer(current_packet_);
```

Comment:
The stream parser passes the buffer asynchronously so it can not wait after `process()` finishes and free the buffer.

I have assumed, or found very packet processor is wrapped in the async one, yes, and reduced multiple copies of `releaseBuffer()` to just this one.